### DISCUSSION TITLE
Gopro 11 internal color space, gamma correction, color primaries

### DISCUSSION THREAD

**Discussion by IRCGraphic** [gopro/labs#402] [General]
I'm experimenting with oMMtrx and I would like to know if possible what is the camera internal color space, gamma correction, color primaries, and any other color processing algorithms used.

Thanks

**Comment by dnewman-gpsw** [comment]
Rec709 for color space, gamma correction and primaries.  Although Native and Vibrant are enhancements the base of Rec709.  GoPro Flat has the same Rec709 color, with a Log 113 curve as the transfer function (see https://www.desmos.com/calculator/sfxqtunzty)

Using !MWIDE=1 will get you the sensor's native gamut.

**Reply by IRCGraphic** [reply]
I'm trying to build a tool to simulate the MMTRX code with a sample image, the reason being is proving to be very fiddly to control all the parameters of this tool (a 3x3 matrix is no joke in terms of the magnitude of possible color transformations) through the gopro and qr interface, the battery gets wasted in the trial error fast and changing its parameters through the qr interface, smartphone or browser, over and over is sluggish. I'm sure through a some more dynamic way of simulating broadly the color transformations would be much more practical, i've tried it with davinci resolve color space transforms and 3x3 matrix sliders strangely without matching the color response of this tool and also made a script to customise the coefficients and emulate the transforms through an output image sample without matching exactly the MMTRX coefficients response. Could you give any clues to build a tool, or better yet indicate me to an existing tool, that simulates its color transformations in a more practical interface?

For instance I found this coefficients that matches almost perfectly input-output

Formula 1:
Y => 67 , 173 , 15
U => -43 , -84 , 127
V => 127 , -107 , -21

Ultimately my goal to control precisely MMTRX is to combat chromatic noise, common in small sensors, like biasing towards green (cleanest signal), as it's proven with compressed video formats the cleanest way to avoid noise and chroma problems is to cook a "safe look" in-camera taking into account the post-processing colour grading.

Appreciate any help, thanks

**Comment by IRCGraphic** [comment]
Thanks, the MTRX code has different weights: Y= 50 l 175 l 25 What would be the complete matrix (in colour) to achieve an approximative net zero result? It's easier for me to fine tune a colour filter starting with a corrected image from the beginning. 
Thanks

**Reply by dnewman-gpsw** [reply]
Y = 0.212R + 0.715G + 0.072B  Approximate sRGB to Rec709 YCrCb

0.212 x 255 = 54
0.715 x 255 = 182
0.072 x 255 = 18

so 50 | 175 | 25 are just approximations.  

I cannot provide the internal math used, as there might be GoPro IP for look (I have not idea if it does.)

**Reply by IRCGraphic** [reply]
Thanks for your information, don't quite understand what is the IP but I'm guessing camera brands like to keep their beautiful colour science private. Well approximate values but when i desaturate an unaltered video compared to the black and white matrix the results look identical.
Thanks
